src/composables/useBadgeData.js

```diff
@@ -134,8 +134,8 @@ export default function useBadgeData() {
 
 					// Get specific properties used in the UI
 					const completedTiers = sortedTiers.filter(t => !!t.completedDate);
-					const hasStarted = completedTiers.length > 0;
-					const level = hasStarted ? completedTiers[completedTiers.length - 1].level : undefined;
+					const hasStarted = completedTiers.length > 0 || achievementData?.totalProgressToAchievement > 0;
+					const level = hasStarted ? completedTiers?.[completedTiers.length - 1]?.level ?? 1 : undefined;
```

Comment:
The `level` should be `undefined` if a level has not been completed. So I think this line can be reverted.